tests/test_api.py

```diff
@@ -183,7 +191,7 @@ def test_sign_verify(self):
         # Test verifying with explicitly set serializer
         targets_key.verify_signature(md_obj, CanonicalJSONSerializer())
         with self.assertRaises(exceptions.UnsignedMetadataError):
-            targets_key.verify_signature(md_obj, JSONSerializer())
+            targets_key.verify_signature(md_obj, JSONSerializer())  # type: ignore[arg-type]
```

Comment:
Here we receive this warning:
```
tests/test_api.py:194: error: Argument 2 to "verify_signature" of "Key" has incompatible
type "JSONSerializer"; expected "Optional[SignedSerializer]"  [arg-type]
```

I disabled it because the test wanted to test setting an explicit set serializer and which will raise
an `UnsignedMetadataError` and that's what happens.
Please note that `JSONSerializer` inherits the `MetadataSerializer` and not the `SignedSerializer` (as for example the `CanonicalJSONSerializer` does).
If we replace the `JSONSerializer` with the `CanonicalJSONSerializer` the test no longer passes
as we do that same thing on line 192.

Reply:
isn't this a sign of a bug somewhere then (presumably in the annotations since everything seems to work)?

Reply:
oh... you mean the test explicitly wants to use a serializer argument that is a completely different object... but that happens to have a method of same name `serialize()`.

That's a weird test but not a bug then